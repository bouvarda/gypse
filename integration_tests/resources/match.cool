
class ClassA(var a: Int) {}
class ClassB(var b1: String, var b2: String) {}
class ClassC() {}

class Main() {
    var test_int: Int = 1;
    var test_string: String = "test";
    var test_a: ClassA = new ClassA(123);
    var test_b: ClassB = new ClassB("un", "deux");
    var test_c: ClassC = new ClassC();

    def print(obj: Any): Unit = {
        obj match {
            //case i: Int => out_int(i)
            //case s: String => out_string(s)
            case a: ClassA => {
                out_string("A ");
                out_int(a.a);
            }
            case b: ClassB => {
                out_string("B ");
                out_string(b.b1);
                out_string(" ");
                out_string(b.b2);
            }
            case c: ClassC => out_string("C")
            case other: Any => out_string("Not printable")
        };
        out_nl();
    };

    {
        //print(test_int);
        //print(test_string);
        print(test_a);
        print(test_b);
        print(test_c);
    };
}