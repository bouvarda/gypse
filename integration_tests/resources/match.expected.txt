A 123
B un deux
C
